src/Symfony/Bundle/TwigBundle/ErrorRenderer/TwigHtmlErrorRenderer.php

```diff
@@ -0,0 +1,111 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\TwigBundle\ErrorRenderer;
+
+use Symfony\Component\ErrorCatcher\ErrorRenderer\HtmlErrorRenderer;
+use Symfony\Component\ErrorCatcher\Exception\FlattenException;
+use Symfony\Component\HttpFoundation\RequestStack;
+use Symfony\Component\HttpFoundation\Response;
+use Twig\Environment;
+use Twig\Error\LoaderError;
+use Twig\Loader\ExistsLoaderInterface;
+
+/**
+ * @author Yonel Ceruto <yonelceruto@gmail.com>
+ */
+class TwigHtmlErrorRenderer extends HtmlErrorRenderer
+{
+    private $twig;
+    private $requestStack;
+    private $debug;
+
+    public function __construct(Environment $twig, RequestStack $requestStack, bool $debug = true, string $charset = null, $fileLinkFormat = null)
+    {
+        $this->twig = $twig;
+        $this->requestStack = $requestStack;
+        $this->debug = $debug;
+
+        parent::__construct($debug, $charset, $fileLinkFormat);
+    }
+
+    public function render(FlattenException $exception): string
+    {
+        if (null === $request = $this->requestStack->getCurrentRequest()) {
+            return parent::render($exception);
+        }
+
+        $statusCode = $exception->getStatusCode();
+
+        if ($request->attributes->get('showException', $this->debug)) {
```

Comment:
should the full exception rendering used in Debug mode even be rendered using TwigBundle ? Or should we handle that part in ErrorCatcher ? this template is not really the one people would customize, as it is shown only in dev.

Reply:
> should the full exception rendering used in Debug mode even be rendered using TwigBundle ?

Yes, I think so.

> this template is not really the one people would customize, as it is shown only in dev.

I can prefix it with `!` to always load the bundle template and make it unavailable by default.

Reply:
Also the full exception templates provided by TwigBundle is more advanced than the one provided by the ErrorCatcher

Reply:
that's my point. Should we keep maintaining 2 of them, or should we put the full feature set in ErrorCatcher directly, and have TwigBundle provide a template-based extension point only for the prod error page ?

Reply:
And to see the customized HTML error template, one can use the preview mechanism `/_error/404.html`

Reply:
what does the preview mechanism has to do in this discussion ? I know about it, but that's unrelated to my point.

My suggestion is to move the responsibility of the dev-only debugging rendering of error to ErrorCatcher directly (and deprecate this `exception_full.html.twig` template) to maintain only one rendering for it.

Reply:
> what does the preview mechanism has to do in this discussion ? I know about it, but that's unrelated to my point.

Because we need to find a way to switch to debug = false when the preview mechanism is used and show the prod version of the HTML error template. Currently, the ErrorCatcher is not aware of that.

Reply:
> My suggestion is to move the responsibility of the dev-only debugging rendering of error to ErrorCatcher directly (and deprecate this exception_full.html.twig template) to maintain only one rendering for it.

I agree with that even if it's harder to maintain without Twig features (block, include, etc)